integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/RefreshCarbonTableCommand.scala

```diff
@@ -0,0 +1,264 @@
+                        s" not copied under database [$databaseName]"
+              LOGGER.audit(msg)
+              CarbonException.analysisException(msg)
+            }
+            // 2.2.1 Register the aggregate tables to hive
+            registerAggregates(databaseName, dataMapSchemaList, databaseLocation)(sparkSession)
+          }
+          registerTableWithHive(databaseName, tableName, tableInfo)(sparkSession)
+        } else {
+          LOGGER.audit(
+            s"Table registration with Database name [$databaseName] and Table name [$tableName] " +
+            s"failed." +
+            s"Table [$tableName] either non carbon table or stale carbon table under database " +
+            s"[$databaseName]")
+        }
+      } else {
+        LOGGER.audit(
+          s"Table registration with Database name [$databaseName] and Table name [$tableName] " +
+          s"failed." +
+          s"Table [$tableName] either already exists or registered under database [$dbName]")
+      }
+    }
+    finally {
+      // release lock after command execution completion
+      CommonUtil.releaseLocks(locks)
+    }
+    // update the schema modified time
+    metaStore.updateAndTouchSchemasUpdatedTime()
+    Seq.empty
+  }
+
+  /**
+   * the method prepare the data type for raw column
+   *
+   * @param column
+   * @return
+   */
+  def prepareDataType(column: ColumnSchema): String = {
+    column.getDataType.getName.toLowerCase() match {
+      case "decimal" =>
+        "decimal(" + column.getPrecision + "," + column.getScale + ")"
+      case others =>
+        others
+    }
+  }
+
+  /**
+   * The method register the carbon table with hive
+   *
+   * @param dbName
+   * @param tableName
+   * @param tableInfo
+   * @param sparkSession
+   * @return
+   */
+  def registerTableWithHive(dbName: String,
+      tableName: String,
+      tableInfo: TableInfo)(sparkSession: SparkSession): Any = {
+    val carbonMetaStore = CarbonEnv.getInstance(sparkSession).carbonMetastore
+    val carbonSchemaString = carbonMetaStore.generateTableSchemaString(tableInfo,
+      tableInfo.getOrCreateAbsoluteTableIdentifier())
+    val tablePath = tableInfo.getTablePath
+    try {
+      val columns = tableInfo.getFactTable.getListOfColumns.asScala.filter(!_.isInvisible)
+      // TODO adding the complex type fields should be handled
+      // getAll Fields
+      val fields = new Array[String](columns.size)
+      columns.map(column => fields(column.getSchemaOrdinal)
+        = column.getColumnName + ' ' + prepareDataType(column))
+      val operationContext = new OperationContext
+      val createTablePreExecutionEvent: RefreshTablePreExecutionEvent =
+        new RefreshTablePreExecutionEvent(sparkSession,
+          tableInfo.getOrCreateAbsoluteTableIdentifier().getCarbonTableIdentifier,
+          tablePath)
+      OperationListenerBus.getInstance.fireEvent(createTablePreExecutionEvent, operationContext)
+      sparkSession.sql(
```

Comment:
I think you  can directly CreateTableCOmmand here instead of creating sql here.

Reply:
@ravipesala Please elaborate how to use the same i don't see any instance.